functions/taskrouterListeners/createHrmContactListener.private.ts

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+/* eslint-disable global-require */
+/* eslint-disable import/no-dynamic-require */
+import '@twilio-labs/serverless-runtime-types';
+import { Context } from '@twilio-labs/serverless-runtime-types/types';
+
+import {
+  EventFields,
+  EventType,
+  RESERVATION_ACCEPTED,
+  TaskrouterListener,
+} from '@tech-matters/serverless-helpers/taskrouter';
+import { HrmContact, PrepopulateForm } from '../hrm/prepopulateForm';
+
+export const eventTypes: EventType[] = [RESERVATION_ACCEPTED];
+
+type EnvVars = {
+  TWILIO_WORKSPACE_SID: string;
+  CHAT_SERVICE_SID: string;
+  HRM_STATIC_KEY: string;
+};
+
+// Temporarily copied to this repo, will share the flex types when we move them into the same repo
+
+const BLANK_CONTACT: HrmContact = {
+  id: '',
+  timeOfContact: new Date().toISOString(),
+  taskId: null,
+  helpline: '',
+  rawJson: {
+    childInformation: {},
+    callerInformation: {},
+    caseInformation: {},
+    callType: '',
+    contactlessTask: {
+      channel: 'web',
+      date: '',
+      time: '',
+      createdOnBehalfOf: '',
+      helpline: '',
+    },
+    categories: {},
+  },
+  channelSid: '',
+  serviceSid: '',
+  channel: 'default',
+  createdBy: '',
+  createdAt: '',
+  updatedBy: '',
+  updatedAt: '',
+  queueName: '',
+  number: '',
+  conversationDuration: 0,
+  csamReports: [],
+  conversationMedia: [],
+};
+
+/**
+ * Checks the event type to determine if the listener should handle the event or not.
+ * If it returns true, the taskrouter will invoke this listener.
+ */
+export const shouldHandle = ({
+  TaskAttributes: taskAttributesString,
+  TaskSid: taskSid,
+  EventType: eventType,
+}: EventFields) => {
+  if (!eventTypes.includes(eventType)) return false;
+
+  const { isContactlessTask, transferTargetType } = JSON.parse(taskAttributesString ?? '{}');
+
+  if (isContactlessTask) {
+    console.debug(`Task ${taskSid} is a contactless task, contact was already created in Flex.`);
+    return false;
+  }
+
+  if (transferTargetType) {
```

Comment:
Maybe this could reuse the `hasTransferStarted` method (`functions/transfer/helpers.private.ts`)?

Reply:
Not worth pulling it in for IMHO, when it's a lambda and we can import stuff properly then sure